### PR TITLE
Add references for fourslash in test dir

### PR DESCRIPTION
**Pull request by aozgaa**
We want to expose `src/harness/fourslash.ts` to `tests/cases/fourslash/fourslash.ts`, which defines the API/DSL for 4/ tests. However, we don't want to expose `src/harness/fourslash.ts` (or the rest of the compiler) to the tests themselves, since this crowds the `ts` namespace. Up to this point, we have decided to forgo the former, which makes contributing to the API difficult and error-prone (we give up static typing!).

As a temporary measure, I've added instructions on how to toggle between editing the API and using it.

One immediate issue: should these instructions immediately follow the 4/ guide? It may be somewhat disheartening to see the guide immediately followed by a hacky solution.

### CODE REVIEW

File: tests/cases/fourslash/fourslash.ts [hunk 1]
```diff
@@ -29,7 +29,34 @@
...
+// For API editors:
+// When editting this file, and only while editing this file, enable the reference comments
+// and comment out the declarations in this section to get proper type information.
+// Undo these changes before compiling/committing/editing any other fourslash tests.
+// The test suite will likely crash if you try 'jake runtests' with reference comments enabled.
+//
+// Explanation:
+// We want type-completion while we edit this file, but at compile time/while editting fourslash tests,
+// we don't want to include the following reference because we are compiling this file in "--out" mode and don't want to rope
+// in the entire codebase into the compilattion each fourslash test. Additionally, we don't want to expose the
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/fourslash.ts]: compilation

File: src/harness/fourslash.ts [hunk 2]
```diff
@@ -1875,7 +1875,7 @@ module FourSlash {
             }
         }
 
-        private verifyProjectInfo(expected: string[]) {
+        public verifyProjectInfo(expected: string[]) {
```
Comment (DanielRosenwasser) [inline src/harness/fourslash.ts]: Why `public`? I'm not seeing any change in usage.

Comment (aozgaa) [inline src/harness/fourslash.ts]: `verifyProjectInfo()` is used at https://github.com/Microsoft/TypeScript/blob/addReferencesForFourslashInTestDir/tests/cases/fourslash/fourslash.ts#L468

File: tests/cases/fourslash/fourslash.ts (hunk 1 above)
Comment (aozgaa) [inline tests/cases/fourslash/fourslash.ts]: Fixed.

**Comment by RyanCavanaugh** [comment]
:+1:

**Pull request merged**